src/saml2/entity.py

```diff
@@ -141,6 +143,24 @@ def __init__(self, entity_type, config=None, config_file="",
         else:
             raise SAMLError("Missing configuration")
 
+        def_sig = DefaultSignature()
+        self.signing_algorithm = (
+            self.config.getattr('signing_algorithm')
+            or def_sig.get_sign_alg()
+        )
+        self.digest_algorithm = (
+            self.config.getattr('digest_algorithm')
+            or def_sig.get_digest_alg()
+        )
+        sign_config = (
+            self.config.getattr("authn_requests_signed", "sp")
+            if self.entity_type == "sp"
+            else self.config.getattr("sign_response", "idp")
+            if self.entity_type == "idp"
+            else False
+        )
```

Comment:
````
entity_swtich = {
'sp': self.config.getattr("authn_requests_signed", "sp"),
'idp': self.config.getattr("sign_response", "idp")
}

sign_config = entity_swtich.get(self.entity_type, False)
````

Reply:
That's an old trick for Who think that python doesn't have an seamless switch statement 😂